src/main/java/com/ericsson/ei/mongo/MongoDBHandler.java

```diff
@@ -363,42 +361,62 @@ private boolean doDrop(String dataBaseName, String collectionName, MongoQuery qu
 
     }
 
-    private MongoCollection<Document> getMongoCollection(String dataBaseName,
+    private MongoCollection<Document> getMongoCollection(String databaseName,
             String collectionName) {
-        if (mongoClient == null)
+        if (mongoClient == null) {
             return null;
-        MongoDatabase db;
+        }
+
+        try {
+            List<String> collectionList = getCollectionList(databaseName);
+            if (!collectionList.contains(collectionName)) {
+                createCollection(databaseName, collectionName);
+            }
+        } catch (MongoClientException e) {
+            LOGGER.error("Failure when handling Mongo collection: {} ", e.getMessage(), e);
+            return null;
+        }
+
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
+        MongoCollection<Document> collection = db.getCollection(collectionName);
+        return collection;
+    }
+
+    private List<String> getCollectionList(String databaseName) {
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
         List<String> collectionList;
 
         try {
-            db = mongoClient.getDatabase(dataBaseName);
             collectionList = db.listCollectionNames().into(new ArrayList<String>());
         } catch (MongoInterruptedException | MongoSocketReadException | MongoSocketWriteException
                 | MongoCommandException | IllegalStateException e) {
-            LOGGER.error("Failed to get Mongo collection, Reason : {} ", e.getMessage(), e);
+            String message = "Failed to get Mongo collection list, Reason: " + e.getMessage();
             closeMongoDbConnection();
-            return null;
+            throw new MongoClientException(message, e);
         }
 
-        if (!collectionList.contains(collectionName)) {
-            LOGGER.debug(
-                    "The requested database({}) / collection({}) not available in mongodb, Creating ........",
-                    dataBaseName, collectionName);
-            try {
-                db.createCollection(collectionName);
-            } catch (MongoCommandException e) {
-                String message = "collection '" + dataBaseName + "." + collectionName
-                        + "' already exists";
-                if (e.getMessage().contains(message)) {
-                    LOGGER.warn("A {}.", message, e);
-                } else {
-                    throw e;
-                }
+        return collectionList;
+    }
+
+    private void createCollection(String databaseName, String collectionName) {
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
+        LOGGER.debug("The requested database({}) / collection({}) not found in mongodb, Creating.",
+                databaseName, collectionName);
+        try {
```

Comment:
Please move the code above inside the try catch as we agreed on having try/catch first and last thing in a method.

Reply:
Done in next commit.